snuba/settings/validation.py

```diff
@@ -86,12 +90,21 @@ def validate_settings(locals: Mapping[str, Any]) -> None:
                 # that are not defined in StorageSetKey.
                 pass
 
-    for logical_part in range(0, SENTRY_LOGICAL_PARTITIONS):
-        physical_part = locals["LOGICAL_PARTITION_MAPPING"].get(str(logical_part))
-        partition_count = locals["LOCAL_SLICES"]
+    for storage in locals["SLICED_STORAGES"]:
         assert (
-            physical_part is not None
-        ), f"missing physical slice for logical partition {logical_part}"
-        assert (
-            physical_part < locals["LOCAL_SLICES"]
-        ), f"physical slice for logical partition {logical_part} is {physical_part}, but only {partition_count} physical slices exist"
+            storage in locals["LOGICAL_PARTITION_MAPPING"]
+        ), "sliced mapping must be defined for sliced storage {storage}"
+
+        storage_mapping = locals["LOGICAL_PARTITION_MAPPING"][storage]
+        defined_slice_count = locals["SLICED_STORAGES"][storage]
+
+        for logical_part in range(0, SENTRY_LOGICAL_PARTITIONS):
+            slice_id = storage_mapping.get(logical_part)
+
+            assert (
+                slice_id is not None
+            ), f"missing physical slice for storage {storage}'s logical partition {logical_part}"
+
+            assert slice_id < defined_slice_count, slice_count_validation_msg.format(
```

Comment:
nit: let's change this condition `slice_id >= 0 and slice_id  < defined_slice_count`